### PR TITLE
Adding a command line parameter "-db-host" to optionally specify a re…

### PR DESCRIPTION
**Pull request by kevand900**
…mote machine to host the mysql database.

This option is to be used with "--i-am-not-benchmarking".
Changes were made in PerfOptions.php for the command line parsing, and DatabaseInstaller.php for installing the database.
Drupal7, Mediawiki, and Wordpress are the targets which support this feature presently.

------------------------------------------------
--ON THE SERVER RUNNING MYSQL--

Edit my.cnf
$sudo vim /etc/mysql.cnf

Comment out '#skip-networking' if present:
Change to 'bind-address=0.0.0.0'

Restart the server:
$sudo service mariadb stop
$sudo service mariadb start

Create a user 'root' with IP of the machine running OSS:
MariaDB [(none)]>CREATE USER 'root'@'<SomeIP>' IDENTIFIED BY 'root';

Grant Privileges to this user:
MariaDB [(none)]>GRANT ALL PRIVILEGES ON *.* TO 'root'@'<SomeIP>' IDENTIFIED BY 'root' WITH GRANT OPTION;
MariaDB [(none)]>FLUSH PRIVILEGES;

Make sure there is a user named 'root'@'localhost.localdomain':
MariaDB [(none)]>SELECT User, Host FROM mysql.user;
+----------+-----------------------+
| User     | Host                  |
+----------+-----------------------+
| root     | 127.0.0.1             |
| wp_bench | 127.0.0.1             |
| root     | ::1                   |
| root     | localhost             |
| root     | localhost.localdomain |
+----------+-----------------------+

Allow remote access through IPTables:
# iptables -P INPUT ACCEPT
# iptables -F
# iptables -A INPUT -i lo -j ACCEPT

----------------------------------------

ON MACHINE RUNNING OSS
Test connection from machine running OSS:
$mysql -u root -proot -h <SomeIP>

-----------------------------------------

When running OSS, add the command line parameter "db-host=<SomeIP>" along with "--i-am-not-benchmarking"

-------

### CODE REVIEW

File: base/PerfOptions.php [hunk 1]
```diff
@@ -110,6 +110,8 @@
 
   public bool $notBenchmarking = false;
 
+  public ?string $dbHost; //The hostname/IP of server which hosts the database.
```
Comment (mofarrell) [inline base/PerfOptions.php]: This doesn't need to be optionally null, and if it is null things will break.

File: targets/drupal7/Drupal7Target.php [hunk 2]
```diff
@@ -31,6 +31,15 @@ public function install(): void {
       );
     }
 
+    copy('compress.zlib://' . __DIR__.'/settings.php.gz', __DIR__.'/temp.php');
+
+    $file = __DIR__.'/temp.php';
+    $file_contents = file_get_contents($file);
+    $file_contents = preg_replace('/^.*\'host\' => \'.*$/m', '      \'host\' => \''.$this->options->dbHost.'\',', $file_contents );
```
Comment (mofarrell) [inline targets/drupal7/Drupal7Target.php]: 80 chars per line.  This should happen everywhere.

File: targets/drupal7/Drupal7Target.php [hunk 3]
```diff
@@ -31,6 +31,15 @@ public function install(): void {
       );
     }
 
+    copy('compress.zlib://' . __DIR__.'/settings.php.gz', __DIR__.'/temp.php');
```
Comment (mofarrell) [inline targets/drupal7/Drupal7Target.php]: This section is modifying the files that get checked in.  We don't want to do that if possible.  This should modify the files after they are copied to the tmp directory used for the test.

Also consider putting a pattern in the settings.php file that you match for.  Something like:
`__DB_HOST__`.  You can find other parts of this project that use that style.

File: base/DatabaseInstaller.php [hunk 4]
```diff
@@ -46,8 +48,8 @@ public function installDatabase(): bool {
       $this->checkMySQLConnectionLimit();
       return false;
     }
-
-    $conn = mysql_connect('127.0.0.1', $db, $db);
+
```
Comment (mofarrell) [inline base/DatabaseInstaller.php]: Try not to add whitespace at the end of lines.

File: targets/mediawiki/MediaWikiTarget.php [hunk 5]
```diff
@@ -40,6 +40,11 @@ public function install(): void {
     $cache_dir = $this->getSourceRoot().'/mw-cache';
     mkdir($cache_dir);
 
+    $file = __DIR__.'/LocalSettings.php';
```
Comment (mofarrell) [inline targets/mediawiki/MediaWikiTarget.php]: Just like with drupal7 this should modify the file after it has been copied to the tmp directory where the test is going to be run from.  (And use a pattern)

File: targets/wordpress/WordpressTarget.php [hunk 6]
```diff
@@ -28,6 +28,11 @@ public function install(): void {
       );
     }
 
+    $file = __DIR__.'/wp-config.php';
+    $file_contents = file_get_contents($file);
+    $file_contents = preg_replace('/^.*DB_HOST.*$/m', 'define(\'DB_HOST\', \''.$this->options->dbHost.'\');', $file_contents );
+    file_put_contents($file, $file_contents);
```
Comment (mofarrell) [inline targets/wordpress/WordpressTarget.php]: Again try using a pattern, and modify the temp copy not the checked in files.

File: base/PerfOptions.php [hunk 7]
```diff
@@ -292,6 +295,10 @@ public function __construct(Vector<string> $argv) {
 
     $argTempDir = $this->getNullableString('temp-dir');
 
+    if(array_key_exists('db-host', $o)){
```
Comment (mofarrell) [inline base/PerfOptions.php]: This needs to be an option that requires `--i-am-not-benchmarking`.  Use one of the `getNullableString` functions to make it so.

File: targets/drupal7/Drupal7Target.php [hunk 8]
```diff
@@ -30,11 +30,16 @@ public function install(): void {
         $this->getSourceRoot().'/sites/default',
       );
     }
-
+
```
Comment (mofarrell) [inline targets/drupal7/Drupal7Target.php]: Avoid unneeded spaces here as well.

**Comment by kevand900** [comment]
Can you please review? Thanks!

**Pull request merged**